### PR TITLE
Components E2E test cleanup

### LINKED ISSUE #38613: Component E2E test failure in Ubuntu
Tests:
- Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests.ServerJSRootComponentsTest.CanAddAndDisposeRootComponents
- Microsoft.AspNetCore.Components.E2ETest.Tests.JSRootComponentsTest.CanAddAndDisposeRootComponents

https://dev.azure.com/dnceng/public/_build/results?buildId=1487604&view=ms.vss-test-web.build-test-results-tab&runId=42535622&resultId=100740&paneView=debug

System.InvalidCastException : Unable to cast object of type 'System.Collections.Generic.Dictionary`2[System.String,System.Object]' to type 'OpenQA.Selenium.IWebElement'.

```C#
Stack trace
   at Microsoft.AspNetCore.Components.E2ETest.Tests.JSRootComponentsTest.GetShadowRoot(IWebElement element) in /home/vsts/work/1/s/src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs:line 281
   at Microsoft.AspNetCore.Components.E2ETest.Tests.JSRootComponentsTest.CanAddAndDisposeRootComponents(Boolean intoBlazorUi, Boolean attachShadowRoot) in /home/vsts/work/1/s/src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs:line 57
```

Failing build: https://dev.azure.com/dnceng/public/_build/results?buildId=1482803&view=results

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
The components E2E tests drifted a bit out of sync with reality, so this gets them back into line.

1. **All quarantined tests dealt with**. There were three categories:
 * Those that have actually always passed since the great E2E reliability push many months ago. These are simply unquarantined.
 * One test that became broken because of a Selenium API change. This is fixed in this PR and unquarantined. Fixes https://github.com/dotnet/aspnetcore/issues/38613.
 * The ignitor tests.
2. **Ignitor and its tests removed**. We've been saying for a while we want to do this, since the behavior is covered sufficiently by the Selenium E2E tests and certain relatively new unit tests, and because we had never managed to make Ignitor truly stable. However nobody was ready to delete them so they got left in a limbo "quarantined" state even though we had no plans to unquarantine them ever, and their results didn't give us any value whether they passed or failed. This PR finally puts them out of their misery.

The net result is that nothing in Components E2E test is quarantined after this. My personal preference would be not to even have a "quarantine" concept in this area, since I find it not to add value and results in us just ignoring problems over a long period. I'm not removing the quarantining support in this PR since it was only added relatively recently, but lets see if we can avoid using it in the coming months!

### CODE REVIEW

**Review by TanayParikh — APPROVED** [review]
LGTM.

**Review by pranavkm — APPROVED** [review]

**Review by wtgodbe — APPROVED** [review]
Nice!

File: src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs [hunk 1]
```diff
@@ -50,7 +49,7 @@ public void CanAddAndDisposeRootComponents(bool intoBlazorUi, bool attachShadowR
 
         // They render and work
         var containerId = intoBlazorUi ? "container-rendered-by-blazor" : "root-container-1";
-        var dynamicRootContainer = Browser.FindElement(By.Id(containerId));
+        ISearchContext dynamicRootContainer = Browser.FindElement(By.Id(containerId));
```
Comment (dougbu) [inline src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs:52]: Just curious: What are the changes in this file about❔

**Review by dougbu — COMMENTED** [review]
Side questions:

- What is your objection to quarantining in this context❔ My preference is to use it everywhere, especially because we saw flaky tests fail our PRs (and rolling builds) as soon as they were un-skipped or un-quarantined before we added the 30 day wait to the process.
- Why wasn't #33733 every marked as `test-fixed`❔ Many of these tests would have been unquarantined a while ago if we'd followed the agreed process.

**Comment by dougbu** [comment]
s/their misery/our misery/ 😃

File: src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs (hunk 1 above)
Comment (TanayParikh) [reply src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs:52]: Selenium (breaking) API changes.

**Pull request merged**